### PR TITLE
SY-4236: Add IsValid Method to Oracle-Generated Go Enums

### PR DESCRIPTION
**Pull request by pjdotson**
# Issue Pull Request

## Linear Issue

[SY-4236](https://linear.app/synnax/issue/SY-4236)

## Description

Oracle now generates an `IsValid()` method on each **string** enum it emits for Go. The method returns `true` when the receiver equals one of the defined enum constants and `false` otherwise, giving callers a first-class way to validate an enum value (e.g. one decoded off the wire or supplied by a client) instead of writing ad-hoc membership checks.

```go
// IsValid reports whether n is one of the defined Notation values.
func (n Notation) IsValid() bool {
	switch n {
	case NotationStandard, NotationScientific, NotationEngineering:
		return true
	default:
		return false
	}
}
```

After regenerating (`oracle sync`), the method lands on `notation.Notation`, `telem.TimeZone`, `telem.TimestampFormat`, `status.Variant`, `text.Level`, `access.Action`, `ontology.ResourceType`, `channel.OperationType`, `arc.Mode`, and the `spatial.*` enums.

### Why string enums only

`IsValid` is generated for string enums and **deliberately not** for int enums:

- **Bitflags**: `arc/go/types.ChanDirection` is a bitfield (`d & ChanDirectionRead`), so a combined `Read|Write` (= 3) is a legitimate value that is *not* one of the named constants. A switch over defined constants would wrongly report it invalid.
- **Zero/sentinel values**: several int enums use a zero `Invalid`/`None` sentinel, and some already carry custom methods (`IsSet`, `IsRead`) and `validate.NewInclusiveBoundsChecker`-based validation.

The generator can't distinguish these cases, so emitting `IsValid` for int enums would produce incorrect semantics. String enums are closed, unordered sets where membership *is* validity.

### Changes

- `oracle/plugin/go/types/types.go`: added a `Receiver()` helper on `enumData` and extended the string-enum template branch to emit `IsValid()`.
- `oracle/plugin/go/types/types_test.go`: specs asserting `IsValid` is generated for string enums and absent for int enums.
- Regenerated `.gen.go` files across `x/go/*` and `core/pkg/**` via `oracle sync`.
- `oracle/install.sh`: build Oracle with `-buildvcs=false` so it compiles from a Git worktree (where `.git` is a file, not a directory).

No consumers are wired up to `IsValid` yet — that's intentional. The two genuine Go-side gaps (`channel.OperationType` is decoded from the wire without validation; `arc.Mode` silently falls through to the graph branch on an unknown value) are behavior changes and are left as follow-ups.

## Basic Readiness

- [x] I have performed a self-review of my code.
- [x] I have added relevant, automated tests to cover the changes.
- [ ] I have updated documentation to reflect the changes.

<h3>Greptile Summary</h3>

This PR teaches the Oracle Go code generator to emit an `IsValid()` method on every string-typed enum it generates, then re-runs `oracle sync` to propagate the new method across all affected packages. String enums are chosen deliberately because they form closed, unordered sets where membership equals validity, unlike int enums (bitflags, sentinels).

- **Generator (`oracle/plugin/go/types/types.go`)**: A `Receiver()` helper is added to `enumData` and the string-enum template branch is extended with an `IsValid()` switch block, gated on `{{- if $enum.Values}}` to handle the degenerate empty-enum edge case.
- **Tests (`oracle/plugin/go/types/types_test.go`)**: Two new specs verify that `IsValid` is generated for string enums and absent for int enums, using the `MustSucceed` helper imported from `x/testutil`.
- **Regenerated files**: Ten `.gen.go` files across `x/go/*` and `core/pkg/**` receive the new method; `oracle/install.sh` gains `-buildvcs=false` so Oracle compiles correctly from a Git worktree.

<h3>Confidence Score: 5/5</h3>

This PR is additive only — new methods on existing types, no changed logic in consumers — and is safe to merge.

Every .gen.go change is mechanical output from the updated template; the generator logic is well-tested by the new specs; the install.sh worktree fix is isolated and well-commented. No consumer code is modified, so regressions in existing behaviour are not possible.

No files require special attention. The Receiver() empty-name edge case in oracle/plugin/go/types/types.go is already tracked in the review thread.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| oracle/plugin/go/types/types.go | Adds Receiver() helper on enumData and extends the string-enum template to emit IsValid(); the Receiver() call is guarded by {{- if $enum.Values}} so empty-values enums skip it, but an empty Name would still panic (already flagged in review thread). |
| oracle/plugin/go/types/types_test.go | Adds two specs: one asserting IsValid is generated for string enums (checking receiver letter, method signature, and case list), and one asserting it is absent for int enums. Coverage is complete for the new generator branch. |
| oracle/install.sh | Adds -buildvcs=false to the go build invocation to support builds from Git worktrees; well-commented trade-off (loses embedded git revision, retains BuildTime via ldflags). |
| x/go/spatial/types.gen.go | Regenerated file; IsValid added to all spatial string enums. CenterLocation with its single-value switch is valid Go. |
| core/pkg/distribution/ontology/types.gen.go | Regenerated; IsValid added to ResourceType correctly enumerating all 23 constants on one case line. |

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[Oracle template engine] --> B{IsIntEnum?}
    B -->|Yes| C[Emit uint8 type + iota const block, No IsValid]
    B -->|No - string enum| D{Has Values?}
    D -->|No| E[Emit string type + empty const block, No IsValid]
    D -->|Yes| F[Emit string type + const block + IsValid switch]
    F --> G[enumData.Receiver: e.Name at 0 → first letter lowercased]
    F --> H[switch: case Enum1 Enum2 ... return true, default: return false]
```

<sub>Reviews (2): Last reviewed commit: ["Update docstring"](https://github.com/synnaxlabs/synnax/commit/4c0ead08063728609cab12324666a10d0f266a02) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=34023665)</sub>

### CODE REVIEW

File: oracle/plugin/go/types/types.go [hunk 1]
```diff
@@ -495,6 +495,10 @@ type enumData struct {
 	StartsAtOne bool
 }
 
+// Receiver returns the single-letter, lowercased method receiver name for the enum type
+// (e.g. "n" for Notation), falling back to "e" for an empty name.
+func (e enumData) Receiver() string { return strings.ToLower(e.Name[:1]) }
```
[the remark this answers is not in the document]
Comment (pjdotson) [reply oracle/plugin/go/types/types.go:500]: This is fine. Not that this type of fallback coding is a general mark of AI slop and you should be flagging when we write fallbacks in PRs instead of advocating for them.

**Review by emilbon99 — APPROVED** [review]
